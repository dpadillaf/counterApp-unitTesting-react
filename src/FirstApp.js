import React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({ hello, subtitle }) => {

    return (
        <>
            <h1> { hello } </h1>
            <p>{ subtitle }</p>
        </>
    );
}

FirstApp.propTypes = {
    hello: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    subtitle: 'my first app react'
}

export default FirstApp;