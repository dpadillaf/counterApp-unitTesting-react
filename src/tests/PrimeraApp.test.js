import React from 'react';
import FirstApp from "../FirstApp";
import { shallow } from 'enzyme';


describe('test in FirstApp', () => {
    
    // test('should mostrtar mensaje ', () => {
        
    //     const hello="hello, falkry";
    //     const { getByText } = render( <FirstApp hello={ hello } /> );

    //     expect( getByText( hello ) ).toBeInTheDocument();

    // })

    test('should debe de mostrar FirstApp correctamente', () => {
        
        const hello = 'hello, falkry';
        const wrapper = shallow(<FirstApp hello={ hello } />);

        expect( wrapper ).toMatchSnapshot()

    })
    
    test('should mostrar subtitulo enviado por props', () => {
        
        const hello = 'hello, falkry';
        const subtitle = 'i am a subtitle';
        const wrapper = shallow(<FirstApp hello={ hello } subtitle={ subtitle } />);

        const textP = wrapper.find( 'p' ).text();
        expect( textP ).toBe( subtitle );

    })
    

})
