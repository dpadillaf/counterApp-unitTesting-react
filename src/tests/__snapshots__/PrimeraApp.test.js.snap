// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in FirstApp should debe de mostrar FirstApp correctamente 1`] = `
<Fragment>
  <h1>
     
    hello, falkry
     
  </h1>
  <p>
    my first app react
  </p>
</Fragment>
`;
